pkg/gatewayserver/io/ws/lbslns/timesync_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lbslns
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/gatewayserver/io"
+	"go.thethings.network/lorawan-stack/v3/pkg/gatewayserver/io/ws"
+	"go.thethings.network/lorawan-stack/v3/pkg/gatewayserver/scheduling"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/types"
+	"go.thethings.network/lorawan-stack/v3/pkg/util/test"
+	"go.thethings.network/lorawan-stack/v3/pkg/util/test/assertions/should"
+)
+
+func TestTransferTime(t *testing.T) {
+	a, ctx := test.New(t)
+
+	session := &ws.Session{}
+	ctx = ws.NewContextWithSession(ctx, session)
+
+	gtw := &ttnpb.Gateway{
+		Ids: &ttnpb.GatewayIdentifiers{
+			GatewayId: "eui-1122334455667788",
+			Eui:       &types.EUI64{0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88},
+		},
+		FrequencyPlanId: test.EUFrequencyPlanID,
+	}
+
+	conn, err := io.NewConnection(ctx, nil, gtw, test.FrequencyPlanStore, true, nil)
+	if !a.So(err, should.BeNil) {
+		t.FailNow()
+	}
+
+	f := (*lbsLNS)(nil)
+
+	// No time sync available.
+	b, err := f.TransferTime(ctx, time.Now(), conn)
+	if !a.So(err, should.BeNil) {
```

Comment:
Nit; `a.So(err, should.NotBeNil)` is easier to read.

Reply:
The double negation is intentional. The error _should_ be `nil`. If we use `a.So(err, should.NotBeNil)`, the test fails even though the assertion is true.